liquibase-service/src/main/resources/db/rdb_modern/routines/005-sp_nrt_investigation_postprocessing-002.sql

```diff
@@ -0,0 +1,643 @@
+               ,@package_name
+               ,'START'
+               ,@proc_step_no
+               ,@proc_step_name
+               ,@rowcount
+               ,LEFT(@id_list, 500));
+
+
+        SET @proc_step_name = 'Insert into CONFIRMATION_METHOD';
+        SET @proc_step_no = 4;
+
+        -- generate new CONFIRMATION_METHOD_KEY for the corresponding cd
+        delete from dbo.nrt_confirmation_method_key;
+
+        insert into dbo.nrt_confirmation_method_key(confirmation_method_cd)
+        select distinct cmt.confirmation_method_cd
+        from #temp_cm_table cmt
+        where cmt.CONFIRMATION_METHOD_KEY is null
+          and not exists (select confirmation_method_cd
+                          from dbo.confirmation_method cd
+                          where cd.confirmation_method_cd = cmt.confirmation_method_cd);
+
+        /* Insert confirmation_method */
+        insert into dbo.confirmation_method(CONFIRMATION_METHOD_KEY,CONFIRMATION_METHOD_CD,CONFIRMATION_METHOD_DESC)
+        select distinct cmk.D_CONFIRMATION_METHOD_KEY, cmt.confirmation_method_cd, cmt.CONFIRMATION_METHOD_DESC_TXT
+        from #temp_cm_table cmt
+                 join dbo.nrt_confirmation_method_key cmk with (nolock) on cmk.confirmation_method_cd = cmt.confirmation_method_cd
+        where cmt.CONFIRMATION_METHOD_KEY is null
+          and not exists (select confirmation_method_cd
+                          from dbo.confirmation_method cd
+                          where cd.confirmation_method_cd = cmt.confirmation_method_cd);
+
+        /* Logging */
+        set @rowcount = @@rowcount
+        INSERT INTO [dbo].[job_flow_log]
+        (batch_id
+        ,[Dataflow_Name]
+        ,[package_Name]
+        ,[Status_Type]
+        ,[step_number]
+        ,[step_name]
+        ,[row_count]
+        ,[msg_description1])
+        VALUES (@batch_id
+               ,@dataflow_name
+               ,@package_name
+               ,'START'
+               ,@proc_step_no
+               ,@proc_step_name
+               ,@rowcount
+               ,LEFT(@id_list, 500));
+
+        SET @proc_step_name = 'UPDATE CONFIRMATION_METHOD_GROUP';
+        SET @proc_step_no = 5;
+
+        delete dbo.CONFIRMATION_METHOD_GROUP
+        where investigation_key in
+              (select investigation_key from dbo.INVESTIGATION where case_uid in
+                                                                     (select value FROM STRING_SPLIT(@id_list, ','))
+              )
+
+        insert into dbo.CONFIRMATION_METHOD_GROUP ([INVESTIGATION_KEY],[CONFIRMATION_METHOD_KEY],[CONFIRMATION_DT])
+        select cmt.INVESTIGATION_KEY, cm.CONFIRMATION_METHOD_KEY, cmt.CONFIRMATION_DT
+        from #temp_cm_table cmt
+                 left outer join dbo.confirmation_method cm with (nolock) on cmt.confirmation_method_cd = cm.confirmation_method_cd
+
+        /* Need to clear nrt table to support multi select updates */
+        delete dbo.nrt_investigation_confirmation
```

Comment:
Based on the discussion on the 20 second window and updates can come in during that process. Does this delete still hold?

Reply:
it is still needed because a user could unselect all options and click save, in which case we need to clear it out.